internal/subscriber/core.go

```diff
@@ -0,0 +1,54 @@
+package subscriber
+
+import (
+	"context"
+	"strings"
+
+	"github.com/razorpay/metro/internal/brokerstore"
+	"github.com/razorpay/metro/internal/subscription"
+	"github.com/razorpay/metro/pkg/messagebroker"
+	metrov1 "github.com/razorpay/metro/rpc/proto/v1"
+)
+
+// ICore is interface over subscribers core
+type ICore interface {
+	NewSubscriber(ctx context.Context, id string, subscription string, timeoutInSec int, maxOutstandingMessages int64, maxOutstandingBytes int64) (ISubscriber, error)
+}
+
+// Core implements ISubscriber
+type Core struct {
+	bs               brokerstore.IBrokerStore
+	subscriptionCore subscription.ICore
+}
+
+// NewCore returns a new core
+func NewCore(bs brokerstore.IBrokerStore, subscriptionCore subscription.ICore) *Core {
+	return &Core{bs, subscriptionCore}
+}
+
+// NewSubscriber initiates a new subscriber for a given topic
+func (c *Core) NewSubscriber(ctx context.Context, id string, subscription string, timeoutInSec int, maxOutstandingMessages int64, maxOutstandingBytes int64) (ISubscriber, error) {
```

Comment:
can we pass subscription object itself here than all params, most params seems like sub config

Reply:
should not pass the proto subscription, because this layer should not understand proto. Also, not in favour of creating encapsulating structs for args as they pose additional overhead of validating all required options while passing them inline makes it explicit